OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/AdminControllerTests.cs

```diff
@@ -0,0 +1,780 @@
+        
+
+        controller = new AdminController(
+            logger.Object,
+            currentUserService.Object,
+            sensitiveMinistryAdminService.Object,
+            sensitiveApplicationService.Object,
+            sensitiveDirectionService.Object,
+            sensitiveProviderService.Object,
+            localizer.Object)
+        {
+            ControllerContext = new ControllerContext() { HttpContext = httpContext.Object },
+        };
+        providerId = Guid.NewGuid();
+        workshops = FakeWorkshopCards();
+        workshopDto = FakeWorkshop();
+        workshopDto.ProviderId = providerId;
+        children = ChildDtoGenerator.Generate(2).WithSocial(new SocialGroupDto { Id = 1 });
+        providers = ProvidersGenerator.Generate(10);
+        parent = ParentDtoGenerator.Generate().WithUserId(userId);
+        provider = ProviderDtoGenerator.Generate();
+        provider.UserId = userId;
+        provider.Id = providerId;
+        applications = ApplicationDTOsGenerator.Generate(2).WithWorkshopCard(workshops.First()).WithParent(parent);
+        fakeHttpContext = GetFakeHttpContext();
+        controller.ControllerContext.HttpContext = fakeHttpContext;
+    }
+
+    [Test]
+    public async Task GetApplications_WhenCalledByAdmin_ShouldReturnOkResultObject()
+    {
+        // Arrange
+        sensitiveApplicationService.Setup(s => s.GetAll(It.IsAny<ApplicationFilter>())).ReturnsAsync(new SearchResult<ApplicationDto>
+        {
+            Entities = applications,
+            TotalAmount = applications.Count,
+        });
+
+        // Act
+        var result = await controller.GetApplications(new ApplicationFilter()).ConfigureAwait(false) as OkObjectResult;
+
+        // Assert
+        result.Should().NotBeNull();
+        result.StatusCode.Should().Be(StatusCodes.Status200OK);
+    }
+
+    [Test]
+    public async Task GetProviders_WhenNoRecordsInDB_ReturnsNoContentResult()
+    {
+        // Arrange
+        sensitiveProviderService.Setup(x => x.GetByFilter(It.IsAny<ProviderFilter>()))
+            .ReturnsAsync(new SearchResult<ProviderDto> { TotalAmount = 0, Entities = new List<ProviderDto>() });
+
+        currentUserService.Setup(x => x.IsAdmin()).Returns(true);
+
+        // Act
+        var result = await controller.GetByFilterProvider(new ProviderFilter()).ConfigureAwait(false);
+
+        // Assert
+        Assert.IsInstanceOf<NoContentResult>(result);
+    }
+
+    [Test]
+    [TestCase(0)]
+    public void Delete_WhenIdIsInvalid_ReturnsBadRequestObjectResult(long id)
+    {
+        // Arrange
+        sensitiveDirectionService.Setup(x => x.Delete(id));
+
+        // Act and Assert
+        Assert.ThrowsAsync<ArgumentOutOfRangeException>(
+            async () => await controller.DeleteDirectionById(id).ConfigureAwait(false));
+    }
+
+    [Test]
+    public void GetApplications_WhenCalledParentOrProvider_ShouldThrowUnauthorizedAccess()
```

Comment:
(Code readability) Could you rearrange all tests to have them be grouped by same method tests (now it's bit uncomfortable to analyze what tests were written for `GetApplication`, `Delete`, etc. when they are unordered)